### ISSUE TITLE
0.7.0-rc3: thread '' panicked at 'failed to allocate an alternative stack'

### ISSUE THREAD

**Issue by Linicks** [input-output-hk/jormungandr#1054]
**Describe the bug**

***Node crashed with the following log information:***

**  snip std out log  **

Oct 31 18:28:35.677 INFO incoming connection on 167.71.118.1:3000, peer_addr: 95.216.196.132:52302, task: network
Oct 31 18:28:35.677 INFO incoming connection on 167.71.118.1:3000, peer_addr: 85.235.66.63:53490, task: network
Oct 31 18:28:35.678 INFO incoming connection on 167.71.118.1:3000, peer_addr: 198.58.116.105:42736, task: network
Oct 31 18:28:35.678 INFO incoming connection on 167.71.118.1:3000, peer_addr: 188.166.36.69:23322, task: network
Oct 31 18:28:35.678 INFO incoming connection on 167.71.118.1:3000, peer_addr: 13.56.87.134:43568, task: network
Oct 31 18:28:35.678 INFO incoming connection closed, peer_addr: 40.71.93.255:61474, task: network
Oct 31 18:28:35.679 INFO incoming connection closed, peer_addr: 173.212.254.220:40396, task: network
Oct 31 18:28:35.711 INFO incoming connection on 167.71.118.1:3000, peer_addr: 198.58.116.105:42738, task: network
Oct 31 18:28:35.714 INFO incoming connection closed, peer_addr: 146.71.78.59:64253, task: network
Oct 31 18:28:35.714 INFO incoming connection closed, peer_addr: 18.196.168.220:58896, task: network
Oct 31 18:28:35.730 INFO incoming connection closed, peer_addr: 184.169.162.15:35812, task: network
thread 'block0' panicked at 'not yet implemented: method to load a Ref from the storage is not yet there', jormungandr/src/blockchain/chain.rs:259:37
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
not yet implemented: method to load a Ref from the storage is not yet therethread 'leadership0' panicked at 'Expect the event to not close', src/libcore/option.rs:1166:5

**Backtrace:**

stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.34/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.34/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:47
   3: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:36
   4: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:200
   5: std::panicking::default_hook
             at src/libstd/panicking.rs:214
   6: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:477
   7: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:384
   8: std::panicking::begin_panic_fmt
             at src/libstd/panicking.rs:339
   9: <futures::future::either::Either<A,B> as futures::future::Future>::poll
  10: <futures::future::and_then::AndThen<A,B,F> as futures::future::Future>::poll
  11: <futures::future::and_then::AndThen<A,B,F> as futures::future::Future>::poll
  12: <futures::future::and_then::AndThen<A,B,F> as futures::future::Future>::poll
  13: futures::task_impl::Spawn<T>::poll_future_notify
  14: futures::task_impl::std::<impl futures::task_impl::Spawn<F>>::wait_future
  15: jormungandr::blockchain::process::run_handle_input
  16: <futures::stream::for_each::ForEach<S,F,U> as futures::future::Future>::poll
  17: <futures::future::map_err::MapErr<A,F> as futures::future::Future>::poll
  18: futures::task_impl::std::set
  19: tokio_threadpool::worker::Worker::run_task
  20: tokio_threadpool::worker::Worker::run
  21: std::thread::local::LocalKey<T>::with
  22: tokio::runtime::threadpool::builder::Builder::build::{{closure}}
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Oct 31 18:52:27.084 thread 'leadership1' panicked at 'Expect the event to not close', src/libcore/option.rs:1166:5
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.34/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.34/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:47
   3: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:36
   4: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:200
   5: std::panicking::default_hook
             at src/libstd/panicking.rs:214
   6: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:477
   7: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:384
   8: rust_begin_unwind
             at src/libstd/panicking.rs:311
WARN response canceled by the processing task, sub_task: server, task: network
not yet implemented: method to load a Ref from the storage is not yet there

**Mandatory Information**
1. `jcli --full-version` output;
     - jcli 0.7.0-rc3 (HEAD-466c0fb, release, linux [x86_64]) - [rustc 1.38.0 (625451e37 2019-09-23)]
2. `jormungandr --full-version` output;
     - jormungandr 0.7.0-rc3 (HEAD-466c0fb, release, linux [x86_64]) - [rustc 1.38.0 (625451e37 2019-09-23)]

**To Reproduce**

Start the server with the following: $./jormungandr --config ./node-config.yaml --genesis-block-hash ae57995b8fe086ba590c36dc930f2aa9b52b2ffa92c0698fff2347adafe8dc65

Node configuration:
```yaml
---
log:
 format: plain
 level: info
 output: stderr
p2p:
  topics_of_interest:
    blocks: normal
    messages: low
  public_address: "/ip4/<public-ip>/tcp/3000"
  trusted_peers:
   - address: "/ip4/13.230.137.72/tcp/3000"
     id: ed25519_pk1w6f2sclsauhfd6r9ydgvn0yvpvg4p3x3u2m2n7thknwghrfpdu5sgvrql9
   - address: "/ip4/13.230.48.191/tcp/3000"
     id: ed25519_pk1lzrdh0pcmhwcnqdl5cgcu7n0c76pm7g7p6pdey7wup54vz32gy6qlz5vnq
   - address: "/ip4/18.196.168.220/tcp/3000"
     id: ed25519_pk1uufkgu0t9xm8ry04wnddtnku5gjg8typf5z6ehh65uc6nz4j8n4spq0xrl
   - address: "/ip4/3.124.132.123/tcp/3000"
     id: ed25519_pk14tqkqnz3eydn0c8c8gmmyzxgnf2dztpy5dnrx09mhfzv0dh93s3qszqgpc
   - address: "/ip4/18.184.181.30/tcp/3000"
     id: ed25519_pk178ge2jn6c40vgmrewgmg26nmtda47nk2jncukzj327mp3a9g2qzss2d44f
   - address: "/ip4/184.169.162.15/tcp/3000"
     id: ed25519_pk1nk0ne8ez66w5tp2g8ctcakthjpz89eveyg0egcpylenhet83n0sq2jqz8q
   - address: "/ip4/13.56.87.134/tcp/3000"
     id: ed25519_pk1ce450zrtn04eaevcn9csz0thpjuhxrysdrq6qlr9pq7e0wd842nsxy6r5k
rest:
  listen: <private-ip>:3101
storage: "./storage"
```

**Comment by Linicks** [comment]
***Another crash just after deleting the storage directory and restarting the node following the initial bug report.*** 

** tail of log details**

Oct 31 19:20:14.951 INFO connecting to peer, node_id: 1ce2b2e507e3c82a6b3c8939ff0bb2dd2c4da2bdcf2f67828e46968b17546c6d, peer_addr: 94.41.84.85:3100, task: network
Oct 31 19:20:15.107 INFO connection to peer failed, reason: protocol handshake failed: unknown error, node_id: 1c85d1d2684b1e579a2802563299bb1f95915f5336138352363d1da60922b299, peer_addr: 78.11.162.144:3100, task: network
Oct 31 19:20:15.369 INFO failed to connect to peer, reason: Connection refused (os error 111), node_id: 1ce2b2e507e3c82a6b3c8939ff0bb2dd2c4da2bdcf2f67828e46968b17546c6d, peer_addr: 94.41.84.85:3100, task: network
Oct 31 19:20:19.728 INFO disconnecting client, node_id: 93e60b1f116912dbe292913c8eee8b301a45723a0f415e79b6580be56e768f19, peer_addr: 54.39.133.26:3000, task: network
Oct 31 19:20:19.853 INFO peer node ID differs from the expected 0e4ec16658349a48f0e629949bbc34d2fb32391d96a8e0c100d0a9fa7f4faf6d, node_id: 93e60b1f116912dbe292913c8eee8b301a45723a0f415e79b6580be56e768f19, peer_addr: 54.39.133.26:3000, task: network
Oct 31 19:20:26.007thread '<unnamed>' panicked at 'failed to allocate an alternative stack', src/libstd/sys/unix/stack_overflow.rs:134:13
stack backtrace:
thread '<unnamed>' panicked at 'failed to allocate an alternative stack', src/libstd/sys/unix/stack_overflow.rs:134:13
 INFO connecting to peer, node_id: 1e9e552b4f70110f030ae2ecbf101dbac8a51bf0f28b523523034e4a2f3a0bc2, peer_addr: 35.136.128.6:3050, task: network
thread '<unnamed>' panicked at 'failed to allocate an alternative stack', src/libstd/sys/unix/stack_overflow.rs:134:13
Killed

**Comment by NicolasDP** [comment]
this is a duplicate of a known issue #788 

> panicked at 'not yet implemented: method to load a Ref from the storage is not yet there
